src/language-js/print/type-parameters.js

```diff
@@ -27,6 +26,20 @@ import { printTypeScriptMappedTypeModifier } from "./mapped-type.js";
 
 const getTypeParametersGroupId = createGroupIdMapper("typeParameters");
 
+// Keep comma if the file extension not `.ts` and
+// has one type parameter that isn't extend with any types.
+// Because, otherwise formatted result will be invalid as tsx.
+function shouldForceTrailingComma(path, options, paramsKey) {
+  const { node } = path;
+  return (
+    getFunctionParameters(node).length === 1 &&
+    node.type.startsWith("TS") &&
+    !(options.filepath && /\.ts$/.test(options.filepath)) &&
+    !node[paramsKey][0].constraint &&
+    path.parent.type === "ArrowFunctionExpression"
```

Comment:
Could be a micropotimization, but I still wonder if we can move `path.parent.type` further up the list to the reduce the number of times a regex is tested. WDYT?

Reply:
https://github.com/prettier/prettier/pull/14688/commits/332c02c315625b03bab06e492f13408e6436e98a

`path.parent` is a getter, it may not as cheap as you expect, but this is not a hot path, so not really matters.